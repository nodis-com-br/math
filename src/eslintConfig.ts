const rulesTypescript = {
  '@typescript-eslint/explicit-function-return-type': 0,
  '@typescript-eslint/no-explicit-any': 0,
  '@typescript-eslint/no-var-requires': 0,
  'react/prop-types': 0,
  'react/display-name': 0,
  'react/jsx-filename-extension': 0,
  'import/no-extraneous-dependencies': 0,
  'import/prefer-default-export': 0,
  'jsx-a11y/label-has-associated-control': 0,
};

export const typescriptReactNative = {
  extends: ['airbnb', 'prettier', 'prettier/react'],
  plugins: ['react', 'jsx-a11y', 'import', '@typescript-eslint'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 2018,
    sourceType: 'module',
    ecmaFeatures: {
      jsx: true,
    },
  },
  env: {
    jest: true,
  },
  globals: {
    __DEV__: 'readonly',
  },
  rules: rulesTypescript,
  overrides: [
    {
      files: ['*.ts', '*.tsx'],
      rules: {
        '@typescript-eslint/no-unused-vars': [2, { args: 'none' }],
      },
    },
  ],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      node: {
        paths: ['app'],
        extensions: ['.js', '.jsx', '.ts', '.tsx'],
      },
      'babel-module': {
        extensions: ['.js', '.jsx', '.ts', '.tsx'],
      },
    },
  },
};

export const typescriptReactWeb = {
  extends: ['airbnb', 'prettier', 'prettier/react'],
  plugins: ['react', 'jsx-a11y', 'import', '@typescript-eslint', 'cypress'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 2018,
    sourceType: 'module',
    ecmaFeatures: {
      jsx: true,
    },
  },
  env: {
    jest: true,
    'cypress/globals': true,
    browser: true,
    es6: true,
  },
  rules: rulesTypescript,
  overrides: [
    {
      files: ['*.ts', '*.tsx'],
      rules: {
        '@typescript-eslint/no-unused-vars': [2, { args: 'none' }],
      },
    },
  ],
  settings: {
    'import/parsers': {
      '@typescript-eslint/parser': ['.ts', '.tsx'],
    },
    'import/resolver': {
      node: {
        paths: ['src'],
        extensions: ['.js', '.jsx', '.ts', '.tsx'],
      },
    },
  },
};
