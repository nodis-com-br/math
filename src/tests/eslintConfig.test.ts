import { typescriptReactNative, typescriptReactWeb } from '../eslintConfig';

describe('eslintConfig', () => {
  test('typescriptReactNative should match snapshot', () => {
    expect(typescriptReactNative).toMatchSnapshot();
  });

  test('typescriptReactWeb should match snapshot', () => {
    expect(typescriptReactWeb).toMatchSnapshot();
  });
});
