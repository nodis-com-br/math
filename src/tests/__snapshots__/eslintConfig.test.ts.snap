// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eslintConfig typescriptReactNative should match snapshot 1`] = `
Object {
  "env": Object {
    "jest": true,
  },
  "extends": Array [
    "airbnb",
    "prettier",
    "prettier/react",
  ],
  "globals": Object {
    "__DEV__": "readonly",
  },
  "overrides": Array [
    Object {
      "files": Array [
        "*.ts",
        "*.tsx",
      ],
      "rules": Object {
        "@typescript-eslint/no-unused-vars": Array [
          2,
          Object {
            "args": "none",
          },
        ],
      },
    },
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": Object {
    "ecmaFeatures": Object {
      "jsx": true,
    },
    "ecmaVersion": 2018,
    "sourceType": "module",
  },
  "plugins": Array [
    "react",
    "jsx-a11y",
    "import",
    "@typescript-eslint",
    "prettier",
  ],
  "rules": Object {
    "@typescript-eslint/explicit-function-return-type": 0,
    "@typescript-eslint/no-explicit-any": 0,
    "@typescript-eslint/no-var-requires": 0,
    "import/extensions": Array [
      2,
      "ignorePackages",
      Object {
        "js": "never",
        "jsx": "never",
        "ts": "never",
        "tsx": "never",
      },
    ],
    "import/no-extraneous-dependencies": 0,
    "import/prefer-default-export": 0,
    "jsx-a11y/label-has-associated-control": 0,
    "no-param-reassign": 0,
    "prettier/prettier": 2,
    "react/display-name": 0,
    "react/jsx-filename-extension": 0,
    "react/jsx-props-no-spreading": 0,
    "react/prop-types": 0,
  },
  "settings": Object {
    "import/parsers": Object {
      "@typescript-eslint/parser": Array [
        ".ts",
        ".tsx",
      ],
    },
    "import/resolver": Object {
      "babel-module": Object {
        "extensions": Array [
          ".js",
          ".jsx",
          ".ts",
          ".tsx",
        ],
      },
    },
  },
}
`;

exports[`eslintConfig typescriptReactWeb should match snapshot 1`] = `
Object {
  "env": Object {
    "browser": true,
    "cypress/globals": true,
    "es6": true,
    "jest": true,
  },
  "extends": Array [
    "airbnb",
    "prettier",
    "prettier/react",
  ],
  "overrides": Array [
    Object {
      "files": Array [
        "*.ts",
        "*.tsx",
      ],
      "rules": Object {
        "@typescript-eslint/no-unused-vars": Array [
          2,
          Object {
            "args": "none",
          },
        ],
      },
    },
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": Object {
    "ecmaFeatures": Object {
      "jsx": true,
    },
    "ecmaVersion": 2018,
    "sourceType": "module",
  },
  "plugins": Array [
    "react",
    "jsx-a11y",
    "import",
    "@typescript-eslint",
    "cypress",
    "prettier",
  ],
  "rules": Object {
    "@typescript-eslint/explicit-function-return-type": 0,
    "@typescript-eslint/no-explicit-any": 0,
    "@typescript-eslint/no-var-requires": 0,
    "import/extensions": Array [
      2,
      "ignorePackages",
      Object {
        "js": "never",
        "jsx": "never",
        "ts": "never",
        "tsx": "never",
      },
    ],
    "import/no-extraneous-dependencies": 0,
    "import/prefer-default-export": 0,
    "jsx-a11y/label-has-associated-control": 0,
    "no-param-reassign": 0,
    "prettier/prettier": 2,
    "react/display-name": 0,
    "react/jsx-filename-extension": 0,
    "react/jsx-props-no-spreading": 0,
    "react/prop-types": 0,
  },
  "settings": Object {
    "import/parsers": Object {
      "@typescript-eslint/parser": Array [
        ".ts",
        ".tsx",
      ],
    },
    "import/resolver": Object {
      "node": Object {
        "extensions": Array [
          ".js",
          ".jsx",
          ".ts",
          ".tsx",
        ],
        "paths": Array [
          "src",
        ],
      },
    },
  },
}
`;
